app/controllers/registrations_controller.rb

```diff
@@ -194,39 +182,11 @@ def do_update(require_password = true, confirm = false)
       attrs = update_params
       attrs = handle_org(attrs: attrs)
 
-      # user is changing email or password
-      if require_password
-        # if user is changing email
-        if current_user.email == attrs[:email]
-          # remove the current_password because its not actuallyt part of the User record
-          attrs.delete(:current_password)
-
-          # This case is never reached since this method when called with
-          # require_password = true is because the email changed.
-          # The case for password changed goes to do_update_password instead
-          successfully_updated = current_user.update_without_password(attrs)
-        elsif attrs[:password].blank?
-          # password needs to be present
-          message = _('Please enter your password to change email address.')
-          successfully_updated = false
-        elsif current_user.valid_password?(attrs[:current_password])
-          successfully_updated = current_user.update_with_password(attrs)
-          # rubocop:disable Metrics/BlockNesting
```

Comment:
> Do we need to leave this part to handle local user system password changes?

Good question. I'm looking at the following following if-else section in `app/controllers/registrations_controller.rb` prior to this update:

```ruby
      if params[:skip_personal_details] == 'true'
        do_update_password(current_user, update_params)
      else
        do_update(needs_password?(current_user))
      end
```

Also, I'm looking at the following in `app/views/devise/registrations/_password_details.html.erb` (the only other file that uses `:skip_personal_details`:
```ruby
<%= hidden_field_tag :skip_personal_details, "true" %>
```
So whenever the user is wanting to update their password, `do_update_password(current_user, update_params)` will be executed.

I also removed [the following lines](https://github.com/portagenetwork/roadmap/pull/917/files#diff-99d7bf3698ba31b7aedf0f330940825d7f298fe8a94391d8e08281c8c64078d1L197-L206), but the comment further points out how a section of the code will actually never be entered:

```ruby
      # user is changing email or password
      if require_password
        # if user is changing email
        if current_user.email == attrs[:email]
          # remove the current_password because its not actuallyt part of the User record
          attrs.delete(:current_password)

          # This case is never reached since this method when called with
          # require_password = true is because the email changed.
          # The case for password changed goes to do_update_password instead
```
I am a little confused by the `# if user is changing email` comment though. I'm thinking `if current_user.email == attrs[:email]` then they are NOT changing the email.